### PR TITLE
Porting to KituraNIO

### PR DESCRIPTION
**Pull request by saiHemak**
Porting KituraCouchDB to Kitura-NIO

## Description
Modified KituraCouchDB package to use Kitura-NIO instead of KituraNet

## Motivation and Context
To check the compatibility with Kitura-NIO package

## How Has This Been Tested?
Ran Kitura-CouchDB tests

## Checklist:
- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by saiHemak** [comment]
@djones6 KituraNIO uses CNIOOpenSSL library which is dependent upon a lib-ressl native library .XCode requires lib-ressl library to be update for Kitura-NIO .Looks like the failure is due to the same. I am not sure how can I do that in Travis environment .Can you please help me out on the same.

**Comment by saiHemak** [comment]
@djones6  please review

File: Package.swift [hunk 1]
```diff
@@ -31,15 +31,15 @@ let package = Package(
...
+        .package(url: "https://github.com/IBM-Swift/Kitura-NIO.git", from: "0.0.0"),
         .package(url: "https://github.com/IBM-Swift/SwiftyJSON.git", from: "17.0.0"),
     ],
     targets: [
         // Targets are the basic building blocks of a package. A target defines a module or a test suite.
         // Targets can depend on other targets in this package, and on products in packages which this package depends on.
         .target(
             name: "CouchDB",
-            dependencies: ["HeliumLogger", "KituraNet", "SwiftyJSON"]),
+            dependencies: ["HeliumLogger", "KituraNIO", "SwiftyJSON"]),
```
Comment (djones6) [inline Package.swift:42]: This will need to be `KituraNet` now that https://github.com/IBM-Swift/Kitura-NIO/pull/72 has been merged.

**Comment by saiHemak** [comment]
@djones6  Please review

File: Package.swift [hunk 2]
```diff
@@ -31,15 +31,15 @@ let package = Package(
         // Dependencies declare other packages that this package depends on.
         // .package(url: /* package url */, from: "1.0.0"),
         .package(url: "https://github.com/IBM-Swift/HeliumLogger.git", from: "1.7.0"),
-        .package(url: "https://github.com/IBM-Swift/Kitura-net.git", from: "2.1.0"),
+        .package(url: "https://github.com/IBM-Swift/Kitura-NIO.git", from: "0.0.0"),
```
Comment (djones6) [inline Package.swift:34]: I think this now needs to be `from: "1.0.0"` as Kitura-NIO has recently been tagged at that major

File: Sources/CouchDB/CouchDBClient.swift [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 import Foundation
 import SwiftyJSON
-import KituraNet
+import KituraNIO
```
Comment (djones6) [inline Sources/CouchDB/CouchDBClient.swift:19]: All these changes should now be unnecessary (due to https://github.com/IBM-Swift/Kitura-NIO/pull/72)

**Review by djones6 — CHANGES_REQUESTED** [review]

**Review by djones6 — CHANGES_REQUESTED** [review]
Thinking more about this, should we be using the `KITURA_NIO` env var trick (as per https://github.com/IBM-Swift/Kitura/pull/1324) to choose which package to depend on?

I think we'd need to do so in order for someone to have a project that depends on both Kitura and Kitura-CouchDB which can be compiled on either Net or NIO.

**Comment by saiHemak** [comment]
@djones6  I agree.. I am done with changes and waiting for merge  of https://github.com/IBM-Swift/Kitura/pull/1324 .

**Pull request closed without merging**